velox/expression/SignatureBinder.cpp

```diff
@@ -253,6 +254,61 @@ bool SignatureBinderBase::tryBind(
   }
 
   const auto& params = typeSignature.parameters();
+
+  // Handle homogeneous row case: row(T, ...)
+  if (typeSignature.isHomogeneousRow() && typeName == "ROW") {
```

Comment:
if isHomogeneousRow() is true, then isn't it redundant to check typeName == "ROW"?